src/Services/UpdateRecords/enumerateData.service.ts

```diff
@@ -0,0 +1,19 @@
+type UpdateOrCreateFunction<RecordType, CompareType> = (
+    comparisonObject: any,
+    record: RecordType,
+    compareValues: CompareType
+) => Promise<void>;
+
+export const enumerateData = <RecordType, CompareType>(
+    records: RecordType[],
+    compareValues: CompareType,
+    updateOrCreateData: UpdateOrCreateFunction<RecordType, CompareType>,
+    generateComparisonObject: (record: RecordType) => any
+) => {
+    if (records && records.length > 0) {
+        records.forEach(async (record: RecordType) => {
+            const comparisonObject = generateComparisonObject(record);
+            await updateOrCreateData(comparisonObject, record, compareValues);
+        });
+    }
+};
```

Comment:
General is good practice in some tricky logic, to add comments about params. Here it's how it looks
![image](https://github.com/UniversityOfGdanskProjects/API-KIOSK/assets/72080205/14b137a0-7af9-4d31-913b-ba0ff3834a62)
With such solution, If You hover on function in IDE then already you can already get a information about params and what those params do. It makes development faster for other coworkers 🐫